hw/ip_templates/otp_ctrl/dv/env/seq_lib/otp_ctrl_zeroization_vseq.sv.tpl

```diff
@@ -0,0 +1,276 @@
+                         part["read_lock"] == "CSR"]
+write_locked_digest_parts = [part for part in otp_mmap["partitions"] if
+                             part["write_lock"] == "Digest"]
+zeroizable_parts = [part for part in otp_mmap["partitions"] if
+                             part["zeroizable"]]
+buf_parts_without_lc = [part for part in otp_mmap["partitions"] if
+                        part["variant"] == "Buffered"]
+secret_parts = [part for part in otp_mmap["partitions"] if
+                part["secret"]]
+## Partitions + LCI + DAI
+num_err_code = len(otp_mmap["partitions"]) + 2
+%>\
+
+// `define PART_CONTENT_RANGE(i) ${"\\"}
+//     {[PART_BASE_ADDRS[``i``]: PART_OTP_SPECIALS_OFFSETS[``i``] - 1]}
+
+// Digest fields can be zeroized, but the zeroization field should be written to only when the
+// partition is designated to be zeroized.
+// The last 8 bytes or any partition is the zeroizible field, hence ignore it when generating legal
+// addresses to zeroize.
+`define PART_CONTENT_RANGE(i) ${"\\"}
+    {[PartInfo[``i``].offset: PartInfo[``i``+1].offset - 9]}
+
+class otp_ctrl_zeroization_vseq extends otp_ctrl_smoke_vseq;
+  `uvm_object_utils(otp_ctrl_zeroization_vseq)
+  `uvm_object_new
+
+
+  rand bit set_dai_regwen;
+  rand bit enable_trigger_checks;
+
+  bit dut_init_completed;
+
+  constraint regwens_c {
+    set_dai_regwen dist {0 :/ 8, 1 :/ 2};
+  }
+
+  constraint trigger_checks_c {
+    // enable_trigger_checks dist {0 :/ 4, 1 :/ 6};
+    enable_trigger_checks == 0;
+  }
+
+  constraint intr_enable_c {
+    // intr_operation_done prediction in the model needs to be revisited
+    // for now disabling operation_done checks with zeroization and enabling only otp_error
+    en_intr == 2'b10;
+  }
+
+  constraint num_trans_c {
+    num_trans  == 1;
+    num_dai_op inside {[1:10]};
+  }
+
+  constraint partition_index_c {
+    part_idx inside {
+% for part in parts_without_lc:
+      ${Name.to_camel_case(part["name"])}Idx${"" if loop.last else ","}
+% endfor
+    };
+  }
+
+  constraint dai_wr_legal_addr_c {
+% for part in parts_without_lc:
+    if (part_idx == ${Name.to_camel_case(part["name"])}Idx)
+        dai_addr inside `PART_CONTENT_RANGE(${Name.to_camel_case(part["name"])}Idx);
+% endfor
+    solve part_idx before dai_addr;
+  }
+
+
+  task body();
+    bit [TL_DW-1:0] reg_rd_val, rdata0, rdata1, rdata0_pre, rdata1_pre;
+    bit [TL_DW-1:0] addr;
+
+    bit zeroized_partition [0:LifeCycleIdx-1];
+    bit used_zeroized_addrs[bit [TL_DW-1:0]];
+
+    bit dai_regwen_state = 1;
+
+    for (int i = 0; i < NumPart; i++) begin
+      // part_idx = otp_ctrl_part_pkg::part_idx_e'(i);
+      part_idx = i;
+      `uvm_info(`gfn, $sformatf("part_idx :%s, addr offset :%x",
+                                 part_idx.name, PartInfo[i].offset), UVM_LOW);
+    end
+
+    // Adding a mechanism to ignore dai_idle checks in main smoke part of the test.
+    ignore_dai_idle_check = 1;
```

Comment:
This depends on how you resolve dai_rd and dai_wr checks per the comments there.